### PR TITLE
allow createDimension to accept Dimension instances (issue #1145)

### PR DESCRIPTION
**Pull request by jswhit**

### CODE REVIEW

File: src/netCDF4/_netCDF4.pyx [hunk 1]
```diff
@@ -2757,6 +2758,17 @@ is the number of variable dimensions."""
             group = self
         else:
             group = self.createGroup(dirname)
+        # if dimensions is a single string or Dimension instance, 
+        # convert to a tuple.
+        # This prevents a common error that occurs when
+        # dimensions = 'lat' instead of ('lat',)
+        if type(dimensions) == str or type(dimensions) == bytes or\
```
Comment (zerothi) [inline src/netCDF4/_netCDF4.pyx:2765]: why not just `isinstance(dimensions, (str, bytes, Dimension))`? type forces the type but not sub-classes?

File: src/netCDF4/_netCDF4.pyx [hunk 2]
```diff
@@ -2653,8 +2653,9 @@ length greater than one are aliases for `str`.
 Data from netCDF variables is presented to python as numpy arrays with
 the corresponding data type.
 
-`dimensions` must be a tuple containing dimension names (strings) that
-have been defined previously using `Dataset.createDimension`. The default value
+`dimensions` must be a tuple containing `Dimension` instances or
```
Comment (zerothi) [inline src/netCDF4/_netCDF4.pyx:2656]: perhaps to clarify that a mix is allowed, `a tuple containing mixed `Dimension`. Perhaps it's just me ;)

**Review by zerothi — COMMENTED** [review]
Great! Thanks!

File: src/netCDF4/_netCDF4.pyx (hunk 1 above)
Comment (jswhit) [reply src/netCDF4/_netCDF4.pyx]: done

File: src/netCDF4/_netCDF4.pyx (hunk 2 above)
Comment (jswhit) [reply src/netCDF4/_netCDF4.pyx]: use and/or instead of or

**Pull request merged**